### PR TITLE
Add native program account constructor

### PR DESCRIPTION
**Pull request by garious**
#### Problem

I want to write unit tests that create executable accounts and currently its done via copypasta.

#### Summary of Changes

* Add a constructor to the native_loader module.  Not sure why the owner is the program id - seems like a security hole, but that's currently what the bank code does.
* Use new constructor to tighten up the bank builtins

### CODE REVIEW

**Comment by mvines** [comment]
> seems like a security hole,

executable accounts aren't writable so we're ok here I think

**Comment by garious** [comment]
@mvines, no tests fail if we let the native_loader own executable accounts. Seems more natural and safer (no need to reason about the runtimes special case for executables). Can you think of any reason why not do it?

File: sdk/src/rewards_transaction.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+use crate::hash::Hash;
+use crate::pubkey::Pubkey;
+use crate::rewards_program;
+use crate::signature::{Keypair, KeypairUtil};
+use crate::system_instruction::SystemInstruction;
+use crate::system_program;
+use crate::transaction::{Instruction, Transaction};
+use bincode::deserialize;
+
+pub struct RewardsTransaction {}
```
Comment (garious) [inline sdk/src/rewards_transaction.rs]: oops, file destined for another PR. I'll boot this.

**Comment by mvines** [comment]
That seems fine to me

**Comment by garious** [comment]
@jackcmay, if `Account::executable` is true, can we assume `owner` is `loader` and therefore remove the `loader` field?

**Comment by jackcmay** [comment]
@garious Yeah that looks reasonable

**Pull request merged**